proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,159 @@
+
+```swift
+// MyScript.swift
+import Foundation
+import UIKit
+@package(url: "http://github.com/author/example.git") import Example
+```
+
+This script can be executable by using `swift run MyScript.swift` which will utilize Swift package managers dependency fetching and resolution logic.
+
+The `@package(...)` syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+```swift
+import Foundation
+import UIKit
+@package(url: "http://github.com/author/example.git") import Example
+```
+
+#### Dependency version requirements
+
+The `@package(..)` annotation would also allow a developer to specify requirements to ensure your scripts are reproducible. The syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+*Revision*
+```swift
+@package(url: "http://github.com/author/example.git", .revision( "065675b3d1364a6f63b94a9c89be2e9ed0a4c3a1")) import Example
+```
+
+*Branch*
+```swift
+@package(url: "http://github.com/author/example.git", .branch("releases/1.0")) import Example
+```
+
+*Exact Version*
+```swift
+@package(url: "http://github.com/author/example.git", .exact("1.0")) import Example
+```
+
+*Range of version*
+```swift
+@package(url: "http://github.com/author/example.git", .from("1.0")) import Example
+```
+
+*Local dependency*
+Absolute path
+```swift
+@package(path: "/Users/author/example") import Example
+```
+
+Relative path
+```swift
+@package(path: "my_project/internal_package") import Example
+```
+
+#### Transitive dependencies 
+
+Package dependencies automatically make their transitive package dependencies available for use.
+
+For an example, lets look at the Package description below for our example dependency and transitive dependency.
+
+Direct dependency
+```swift
+let package = Package(
+    name: "Example",
+    products: [
+        .library(name: "example", targets: ["Example"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/another_author/TransitiveDependency.git", from: "1.0.0"),
+    ],
+)
+```
+
+Transitive dependency
+```swift
+let package = Package(
+    name: "TransitiveDependency",
+    products: [
+        .library(name: "TransitiveDependency", targets: ["TransitiveDependency"]),
+    ],
+    dependencies: [],
+)
+```
+
+With the above Package descriptions, we can import transitive dependencies by explicitly exporting them in the package attribute definition.
+```swift
+import Foundation
+import UIKit
+@package(url: "http://github.com/author/example", products: ["TransitiveDependency"]) import Example 
```

Comment:
This doesn't seem right since a) there's no product called `TransitiveDependency` and b) you shouldn't be able to reference a transitive product.